Frontend/src/Components/Battery/Battery.js

```diff
@@ -0,0 +1,140 @@
+import {Flex, Center, Box, Spacer, Text, useColorMode} from "@chakra-ui/react";
+import RangeRow from "../Shared/RangeRow";
+import RangeBar from "../Shared/RangeBar";
+import CONSTANTS from "../../data-constants.json";
+import getColor from "../Shared/colors";
+
+export default function Battery(props) {
+    const { colorMode } = useColorMode();
+    const headerBg = getColor("header", colorMode);
+    const borderColor = getColor("border", colorMode);
+
+    const fanEnableText = (props.data?.fan_enable[0] ?? -1.0) ? 'ON' : 'OFF';
+    const fanEnableColor = (props.data?.fan_enable[0] ?? -1.0) ? '#89FC4D' : '#F5C2C1';
+
+    const chargeEnableText = (props.data?.charge_enable[0] ?? -1.0) ? 'Enable' : 'Disable';
+    const chargeEnableColor = (props.data?.charge_enable[0] ?? -1.0) ? '#89FC4D' : '#F5C2C1';
+
+    return (
+        <Flex flex="auto" direction="column" overflowY="scroll">
+            <RangeRow
+                dataTitle="Pack Temp"
+                dataValue={props.data?.pack_temp[0] ?? -1.0}
+                dataConstant={CONSTANTS.pack_temp}
+            />
+            <RangeRow
+                dataTitle="Pack Internal Temp"
+                dataValue={props.data?.pack_internal_temp[0] ?? -1.0}
+                dataConstant={CONSTANTS.pack_internal_temp}
+
+                bg={headerBg}
+            />
+            <Center
+                borderTopWidth={1}
+                borderBottomWidth={1}
+                borderRightWidth={1}
+                borderColor={borderColor}
+                lineHeight='1.2em'
+                height='100%'
+            >
```

Comment:
We should have a `bg` prop to use either the normal or error background depending on the value of `fan_speed`. You can reference `BatteryCells/BatteryPack.js` for the `fan_speed` nominal range checks.

Reply:
Two questions: First I set `bg` like this: 

```jsx
bg={((props.data?.fan_speed[0] ?? -1.0) < CONSTANTS.fan_speed.MIN || (props.data?.fan_speed[0] ?? -1.0) > CONSTANTS.fan_speed.MAX) ?
        getColor("errorBg", colorMode) : headerBg}
```

Was this what you were thinking, or should I use 

```jsx
{((props.data?.pack_temp[0] ?? -1.0) < (20 + 5 * CONSTANTS.fan_speed.MAX)) ?
                         Math.floor(((((props.data?.pack_temp[0] ?? -1.0) >= 20) ? props.data?.pack_temp[0] : 15) - 15) / 5) :
                         CONSTANTS.fan_speed.MAX}
```
(line 69 of BatteryPack.js) as the min value in place of what we're getting from the constants file?

Second, I noticed that the min and max I'm getting from the constants file (0 to 6, I think) doesn't match what the data generator produces (which seems to be 0 to 100). Is this an error in one of those or do I need to use that expression from BatteryPack.js to interpret the number I'm getting from the data generator?

Reply:
@jackwherry, it looks like this edit got lost during the debate, but yeah, it looks like you had the right idea. Now that you have `fanSpeedMin`, you can use the following:
```
bg={((props.data?.fan_speed[0] ?? -1.0) < fanSpeedMin || (props.data?.fan_speed[0] ?? -1.0) > CONSTANTS.fan_speed.MAX) ?
        getColor("errorBg", colorMode) : headerBg}
```